specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2023-06-01-preview/containerregistry.json

```diff
@@ -0,0 +1,8032 @@
+          "items": {
+            "type": "string"
+          }
+        },
+        "notificationsList": {
+          "description": "The list of notifications subscription information for the connected registry.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SyncUpdateProperties": {
+      "description": "The parameters for updating the sync properties of the connected registry with its parent.",
+      "type": "object",
+      "properties": {
+        "schedule": {
+          "description": "The cron expression indicating the schedule that the connected registry will sync with its parent.",
+          "type": "string"
+        },
+        "syncWindow": {
+          "format": "duration",
+          "description": "The time window during which sync is enabled for each schedule occurrence. Specify the duration using the format P[n]Y[n]M[n]DT[n]H[n]M[n]S as per ISO8601.",
+          "type": "string"
+        },
+        "messageTtl": {
+          "format": "duration",
+          "description": "The period of time for which a message is available to sync before it is expired. Specify the duration using the format P[n]Y[n]M[n]DT[n]H[n]M[n]S as per ISO8601.",
+          "type": "string"
+        }
+      }
+    },
+    "CredentialSetListResult": {
+      "description": "The result of a request to list credential sets for a container registry.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of credential sets. Since this list may be incomplete, the nextLink field should be used to request the next list of credential sets.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CredentialSet"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next list of credential sets.",
+          "type": "string"
+        }
+      }
+    },
+    "CredentialSet": {
+      "description": "An object that represents a credential set resource for a container registry.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/IdentityProperties",
+          "description": "Identities associated with the resource. This is used to access the KeyVault secrets."
+        },
+        "properties": {
+          "$ref": "#/definitions/CredentialSetProperties",
+          "description": "The properties of the credential set.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "IdentityProperties": {
+      "description": "Managed identity for the resource.",
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "The principal ID of resource identity.",
+          "type": "string",
+          "readOnly": true
```

Comment:
You have two models with the same model name and here has a readOnly principalId the other one https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json#L1946 doesn't have. which will be a problem in SDK generation.  Can you provide a unique name for it ?

same issue in：
1.https://github.com/Azure/azure-rest-api-specs/pull/24426/files#diff-ec8941a9db152dfb094da578bfa7506d87e572746536a97bce808e26fe641711R4818
and 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json#L1950

2.https://github.com/Azure/azure-rest-api-specs/pull/24426/files#diff-ec8941a9db152dfb094da578bfa7506d87e572746536a97bce808e26fe641711R4889 
and 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json#L2013

Reply:
Thank you for your comment @kazrael2119. I get the issue now. We can not rename the property as this property is part of contract between ARM, Managed identity ACR.

From what I understand, the way to solve this is

1. Create a new api version for containerregistry_build,json and make this property readonly.
2. Revert the readonly change in containerregistry.json for api version 2023-06-01-preview

Let me know if their are other options. Ideally I want these property to be readonly if we are already rolling out new api version but do not want to create a new api version just for this change. On the server side we are already threating this property as read-only.

Reply:
Did you treat principalId as readonly at the service backend in the version 2019-06-01-preview ? If yes, we can change  the refer when generated the code to the readonly ones, and claim it as non-breaking, because the previous non-readonly won't work anyway. If no, we will have to revert the readonly change for now.

Reply:
On the server side, we have always treated this as read-only. The issue lies in Swagger, where we neglected to set this property as read-only. We are now aiming to rectify this.

> If yes, we can change the refer when generated the code to the readonly ones

Is this something I need to do, somewhere? I already got this breaking change approved.

Reply:
@jikuma Thank you for the info, I think nothing need from your side.
@kazrael2119 Could you help on the directive in readme.typescipt.md ? Thanks